relay-event-normalization/src/normalize/span/fix_trees.rs

```diff
@@ -0,0 +1,166 @@
+//! Span tree normalization.
+
+use std::collections::BTreeSet;
+use std::mem;
+
+use relay_event_schema::protocol::{Event, TraceContext};
+use relay_protocol::Error;
+
+/// Enforce that every span has a valid parent. If any `parent_span_id` is pointing nowhere, the
+/// span is re-parented onto the root span.
+///
+/// This is to avoid any nasty surprises in the span buffer specifically. Other readers of spans
+/// (such as the frontend's tree component) were already able to deal with detached spans.
+pub fn fix_trees(event: &mut Event) {
+    let Some(ref mut spans) = event.spans.value_mut() else {
+        return;
+    };
+
+    let Some(contexts) = event.contexts.value_mut() else {
+        return;
+    };
+
+    let Some(trace_context) = contexts.get_mut::<TraceContext>() else {
+        return;
+    };
+
+    let Some(root_span_id) = trace_context.span_id.value() else {
+        return;
+    };
+
+    let valid_span_ids = spans
+        .iter()
+        .filter_map(|span| span.value())
+        .filter_map(|span| span.span_id.value())
+        .chain(Some(root_span_id))
+        .cloned()
```

Comment:
I really dislike this. In this expression we access `span_id`, in the for-loop we only access `parent_span_id`. In theory those are mutually exclusive accesses, and should be able to be borrowed at the same time. But it's tricky to convince the compiler of that, and so I have to clone here.

It would be better to make `SpanId` a `u16` anyway...

Reply:
Either this or we serialize like in `ProjectKey` which supports `as_str` and `as_ref::<str>`: 
https://github.com/getsentry/relay/blob/cbf72c2930ad01ed0d780e36968088cd6a051878/relay-base-schema/src/project.rs#L85-L90)

Reply:
Right, part of this would be refactoring all usage to remove access to the field.